src/auth/handlers/oauth.handelr.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpService } from '@nestjs/axios';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Session } from 'src/entities/session.entity';
+import { Repository } from 'typeorm';
+import { User } from '../../entities/user.entity';
+import { EmailHandler } from './mail.handler';
+import { SessionHandler } from './session.handler';
+@Injectable()
+export class OauthHandler {
+  constructor(
+    @InjectRepository(User) 
+    private readonly usersRepository: Repository<User>,
+    private readonly sessionHandler: SessionHandler,
+    private readonly httpService: HttpService
+  ) {}
+    /**
+     * @param  {string} provider
+     * google/facebook
+     * @param  {string} token
+     * access oauth token from the client
+     * @param  {} systemInfo
+     * 
+     */
+    async oauthHandler(provider: string, token: string, systemInfo): Promise<Session>{
+      try {
+        let dataFromProviderURL = '';
+        if (provider === 'facebook'){
+          dataFromProviderURL = process.env.OAUTH_URL_FACEBOOK
+        } else if (provider === 'google') {
+          dataFromProviderURL = process.env.OAUTH_URL_GOOGLE
+        } else {
+          throw new BadRequestException('Provider unavailable')
+        }
+        const userinfo = await this.httpService.axiosRef.get(dataFromProviderURL, {
+          headers : {
+            authorization: 'Bearer ' + token.split(" ")[1]
+        }})
+        return this.authUser(userinfo.data, provider, systemInfo);
+      } catch (error) {
+        console.log(error);
+        throw new BadRequestException('Invalid access token');
+      }
+    }
+
+  /**
+   * @param  {any} profile
+   * user profile from the provider
+   */
+  async authUser(profile: any, provider: string, systemInfo): Promise<Session>{
+    const user = await this.usersRepository
+    .createQueryBuilder('user')
+    .where(`user.oauth ::jsonb @> \'{"id":"${profile.id}"}\'`)
+    .getOne();
+  if(user){
```

Comment:
`if (user) {
    return this.sessionHandler.createSession(user, systemInfo)
}`

remove else condition, just write code after return without `else {}`